applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_coupled_solver.py

```diff
@@ -0,0 +1,277 @@
+            self.__GetDataTransferOperator(data_transfer_operator_name).TransferData(from_solver_data, to_solver_data, i_input_data["data_transfer_operator_options"])
+
+            self.__ExecuteCouplingOperations(i_input_data["after_data_transfer_operations"])
+
+    def _SynchronizeOutputData(self, solver_name):
+        from_solver = self.solver_wrappers[solver_name]
+        output_data_list = self.coupling_sequence[solver_name]["output_data_list"]
+
+        for i in range(output_data_list.size()):
+            i_output_data = output_data_list[i]
+
+            # from solver
+            from_solver_data = from_solver.GetInterfaceData(i_output_data["data"].GetString())
+
+            # to solver
+            to_solver = self.solver_wrappers[i_output_data["to_solver"].GetString()]
+            to_solver_data = to_solver.GetInterfaceData(i_output_data["to_solver_data"].GetString())
+
+            # perform the data transfer
+            self.__ExecuteCouplingOperations(i_output_data["before_data_transfer_operations"])
+
+            data_transfer_operator_name = i_output_data["data_transfer_operator"].GetString()
+            # TODO check the order of solvers!
+            self.__GetDataTransferOperator(data_transfer_operator_name).TransferData(from_solver_data, to_solver_data, i_output_data["data_transfer_operator_options"])
+
+            self.__ExecuteCouplingOperations(i_output_data["after_data_transfer_operations"])
+
+            # Importing data from external solvers
+            from_solver.ExportCouplingInterfaceData(from_solver_data)
+
+
+    def __GetDataTransferOperator(self, data_transfer_operator_name):
+        try:
+            return self.data_transfer_operators_dict[data_transfer_operator_name]
+        except KeyError:
+            raise NameError('The data-transfer-operator "{}" does not exist!'.format(data_transfer_operator_name))
+
+
+    def __ExecuteCouplingOperations(self, settings):
+        for i in range(settings.size()):
+            coupling_operation_name = settings[i].GetString()
+            self.coupling_operations_dict[coupling_operation_name].Execute()
+
+    def PrintInfo(self):
+        super(CoSimulationCoupledSolver, self).PrintInfo()
+
+        cs_print_info(self._Name(), "Has the following components:")
+        for solver in self.solver_wrappers.values():
+            solver.PrintInfo()
+
+        for predictor in self.predictors_list:
+            predictor.PrintInfo()
+
+        for coupling_operation in self.coupling_operations_dict.values():
+            coupling_operation.PrintInfo()
+
+    def Check(self):
+        super(CoSimulationCoupledSolver, self).Check()
+        for solver in self.solver_wrappers.values():
+            solver.Check()
+
+        for predictor in self.predictors_list:
+            predictor.Check()
+
+        for coupling_operation in self.coupling_operations_dict.values():
+            coupling_operation.Check()
+
+    def IsDistributed(self):
+        return True
```

Comment:
Shouldn't this be `False` for the serial version?

Reply:
curently it is not used, I will reevaluate in the next step when I do the MPI-stuff